libbeat/common/transport/tlscommon/tls_config.go

```diff
@@ -151,10 +159,41 @@ func (c *TLSConfig) BuildServerConfig(host string) *tls.Config {
 	return config
 }
 
+func trustESRootCA(cfg *TLSConfig, peerCerts []*x509.Certificate) error {
+	fingerprint, err := hex.DecodeString(cfg.ESCAFingerprint)
+	if err != nil {
+		return fmt.Errorf("decode fingerprint: %w", err)
+	}
+
+	for _, cert := range peerCerts {
+		// Compute digest for each certificate.
+		digest := sha256.Sum256(cert.Raw)
+
+		if bytes.Compare(digest[0:], fingerprint) == 0 {
```

Comment:
This is a bit clearer:

```suggestion
		if bytes.Equal(digest[0:], fingerprint) {
```

Reply:
Done